crates/astria-sequencer/src/app/mod.rs

```diff
@@ -525,11 +525,11 @@ impl App {
         let res = generate_rollup_datas_commitment(&signed_txs_included, deposits);
 
         let txs = match encoded_extended_commit_info {
-            Some(encoded_extended_commit_info) => {
-                std::iter::once(encoded_extended_commit_info.into())
-                    .chain(res.into_iter().chain(included_tx_bytes))
-                    .collect()
-            }
+            Some(encoded_extended_commit_info) => res
+                .into_iter()
+                .chain(std::iter::once(encoded_extended_commit_info.into()))
+                .chain(included_tx_bytes)
+                .collect(),
```

Comment:
An `Option` can be turned into an iterator with 0 or 1 iterations via `into_iter`, so we can shorten this a little:

```rust
        // generate commitment to sequence::Actions and deposits and commitment to the rollup IDs
        // included in the block, chain on the extended commit info if `Some`, and finally chain on
        // the tx bytes.
        let txs = generate_rollup_datas_commitment(&signed_txs_included, deposits)
            .into_iter()
            .chain(encoded_extended_commit_info.map(Bytes::from).into_iter())
            .chain(included_tx_bytes)
            .collect();
```